applications/FluidDynamicsApplication/custom_processes/mass_conservation_check_process.cpp

```diff
@@ -32,129 +32,197 @@ namespace Kratos
 
-    mCorrectionFreq = CorrectionFreq;
-    mWriteToLogFile = WriteToLogFile;
-    mLogFileName = LogFileName;
-    mPerformCorrections = PerformCorrections;
 }
 
 
+
 /// constructor (direct input of settings)
 MassConservationCheckProcess::MassConservationCheckProcess(
     ModelPart& rModelPart,
     Parameters& rParameters)
     : Process(), mrModelPart(rModelPart) {
 
-    Parameters default_parameters( R"(
-    {
-        "model_part_name"                        : "default_model_part_name",
-        "perform_corrections"                    : true,
-        "correction_frequency_in_time_steps"     : 20,
-        "write_to_log_file"                      : true,
-        "log_file_name"                          : "mass_conservation.log"
-    }  )" );
+    const Parameters default_parameters = GetDefaultParameters();
 
     rParameters.ValidateAndAssignDefaults(default_parameters);
 
     mCorrectionFreq = rParameters["correction_frequency_in_time_steps"].GetInt();
-    mWriteToLogFile = rParameters["write_to_log_file"].GetBool();
-    mPerformCorrections = rParameters["perform_corrections"].GetBool();
-    mLogFileName = rParameters["log_file_name"].GetString();
 }
 
 
+
 /// Initialization function to find the initial volumes and print first lines in the log-file
 std::string MassConservationCheckProcess::Initialize(){
 
     double pos_vol = 0.0;
     double neg_vol = 0.0;
     double inter_area = 0.0;
-    const auto& r_comm = mrModelPart.GetCommunicator().GetDataCommunicator();
+    mrModelPart.GetCommunicator().SynchronizeVariable(DISTANCE);
+    mrModelPart.GetCommunicator().SynchronizeVariable(VELOCITY);
+    mrModelPart.GetCommunicator().SynchronizeVariable(MESH_VELOCITY);
 
     ComputeVolumesAndInterface( pos_vol, neg_vol, inter_area );
 
-    this->mInitialPositiveVolume = r_comm.SumAll(pos_vol);
-    this->mInitialNegativeVolume = r_comm.SumAll(neg_vol);
+    this->mInitialPositiveVolume = pos_vol;
+    this->mInitialNegativeVolume = neg_vol;
     this->mTheoreticalNegativeVolume = neg_vol;
-    inter_area = r_comm.SumAll(inter_area);
 
     std::string output_line =   "------ Initial values ----------------- \n";
     output_line +=              "  positive volume (air)   = " + std::to_string(this->mInitialPositiveVolume) + "\n";
     output_line +=              "  negative volume (water) = " + std::to_string(this->mInitialNegativeVolume) + "\n";
     output_line +=              "  fluid interface (area)  = " + std::to_string(inter_area) + "\n\n";
     output_line +=              "------ Time step values --------------- \n";
     output_line +=              "sim_time \t\twater_vol \t\tair_vol \t\twater_err \t\tnet_inf \t\t";
-    output_line +=              "net_outf \t\tint_area \t\tcorr_shift \n";
+    output_line +=              "net_outf \t\tint_area \n";
 
     return output_line;
 }
 
 
-std::string MassConservationCheckProcess::ExecuteInTimeStep(){
+
+std::string MassConservationCheckProcess::ComputeBalancedVolume(){
```

Comment:
Idk if I see using return to build a output. Every part should output its own information and keep the interface as simple as possible.

Reply:
agree too.